src/Membrain/Memory.hs

```diff
@@ -26,13 +28,49 @@ import Data.Ratio (Ratio, (%))
 import GHC.TypeNats (KnownNat, Nat, natVal)
 import Numeric.Natural (Natural)
 
+import Membrain.Units (KnownUnitSymbol, unitSymbol)
 
-{- | Main memory units type. It has phantom parameter 'mem' of kind 'Nat' which
-is type level representation of the unit.
+
+{- | Main memory units type. It has phantom type parameter @mem@ of kind 'Nat'
+which is type level representation of the unit.
 -}
 newtype Memory (mem :: Nat) = Memory
     { unMemory :: Natural
-    } deriving (Show, Eq)
+    } deriving (Eq)
+
+{- | 'Show' instance for 'Memory' shows value as 'Double' with measure unit
+suffix. It shows 'Memory' losslessly while used with standardized units of
+measures. The following math fact is used to display 'Memory'.
+
+A decimal representation written with a repeating final @0@ is said to terminate
+before these zeros. Instead of @1.585000...@ one simply writes @1.585@. The
+decimal is also called a terminating decimal. Terminating decimals represent
+rational numbers of the form  \( \cfrac{k}{2^n 5^m} \).
+
+If you use unconventional units, this operation will hang. TODO: check this statically
```

Comment:
The word `unconventional` here is a bit tricky, is it unconventional for our library, or generally?

Reply:
I will clarify that!